src/Check/DirWritable.php

```diff
@@ -24,30 +20,22 @@
  */
 class DirWritable extends AbstractCheck implements CheckInterface
 {
-    /** @var array|Traversable */
+    /** @var non-empty-list<string> */
     protected $dir;
 
     /**
-     * @param  string|array|Traversable $path Path name or an array of paths
+     * @param  string|non-empty-list<string> $path Path name or an array of paths
```

Comment:
Would a Traversable satisfy this? I do prefer using `list<x>` over `array<x>` myself anyway.